docs/design/vcpkg.md

```diff
@@ -0,0 +1,159 @@
+reside within [vcpkg installation tree][].
+
+vcpkg does not seem to provide any lock file mechanism besides pinning a
+dependency to a specific version.
+
+
+## Specifying dependencies
+
+vcpkg operates with a concept of a port: a versioned recipe for building some
+artifacts. Ports may depend on other ports and may have optional dependencies.
+Each port specifies some metadata about its package including, but not limited
+to name, version and dependencies, location of package source and build
+instructions.
+
+Every port contains a `portfile.cmake` with fetch and build instructions for a
+package and [vcpkg.json][] containing metadata about the package and its
+dependencies. Ports can carry patches to be applied to sources of packages.
+
+<details>
+  <summary>A typical vcpkg.json</summary>
+
+  ```
+  {
+  "name": "fmt",
+  "version": "11.0.2",
+  "port-version": 1,
+  "description": "{fmt} is an open-source formatting library providing a fast and safe alternative to C stdio and C++ iostreams.",
+  "homepage": "https://github.com/fmtlib/fmt",
+  "license": "MIT",
+  "dependencies": [
+      {
+        "name": "vcpkg-cmake",
+        "host": true
+      },
+      {
+        "name": "vcpkg-cmake-config",
+        "host": true
+      }
+    ]
+  }
+  ```
+
+</details>
+
+
+<details>
+  <summary>The relevant part of corresponding portfile.cmake</summary>
+
+  ```
+  vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO fmtlib/fmt
+    REF "${VERSION}"
+    SHA512 47ff6d289dcc22681eea6da465b0348172921e7cafff8fd57a1540d3232cc6b53250a4625c954ee0944c87963b17680ecbc3ea123e43c2c822efe0dc6fa6cef3
+    HEAD_REF master
+    PATCHES
+        fix-write-batch.patch
+        fix-pass-utf-8-only-if-the-compiler-is-MSVC-at-build.patch # remove in next release
+  )
+  ...
+  ```
+
+</details>
+
+Dependencies come from registries which could be of three types:
+
+ 1. The "built-in" registry (effectively a local copy of [vcpkg][] repository);
+ 1. A git registry -- any repository which follows certain layout rules and
+    contains ports;
+ 1. A directory maintaining certain structure.
+
+
+## Downloading dependencies
+
+vcpkg would work on a project created with it. Before using the tool for a new
+project [the official guide][] recommends acquiring vcpkg repository and
+pointing an environment variable to it. This location (`$VCPKG_ROOT`) will be
+used to store sources for dependencies. There is no separate command for just
+downloading sources, however `vcpkg install` accepts `--only-downloads` flag
+which does exactly that. When sources are downloaded they modify two locations:
+vcpkg root directory (effectively a location to which [vcpkg][] repository was
+cloned) and registry cache. The latter defaults to `$HOME/.cache/vcpkg/registries/`
+but could be retargeted with `$X_VCPKG_REGISTRIES_CACHE`. Once both caches are
+populated a package could be built by running `vcpkg install --no-downloads`.
```

Comment:
So, what does a downloaded dependency really look like? I wasn't able to derive that from the docs - the cmake recipes mostly point to Github tarballs/zips of a particular version, does `vcpkg` unpack, patch and repack?

Reply:
From my experiments the sources end up in `buildtrees` subdirectory in $VCPKG_ROOT in unpacked form with applied patches.